### DISCUSSION TITLE
How to decide which engine to use?

### DISCUSSION THREAD

**Discussion by oldwired** [viniciussanchez/RESTRequest4Delphi#120] [Q&A]
RR4D supports RESTClient, Synapse, Indy and NetHTTP as an engine. But what are the pros and cons of those engines? Which one should someone "uninitiated" use?

**Comment by viniciussanchez** [comment]
One might be faster than the other, implement some new features that old components don't have. And that means that if one day a new XYZ component appears, you don't have to rewrite your code. It would suffice to change a compilation directive.

**Reply by oldwired** [reply]
So to start with RR4D I just use the default and if I find it insufficient I look what the others have to offer?

At the moment I mainly use ICS directly, but I like the abstractions and the fluid syntax RR4D offers.  
One thing I have to look is how HTTPS works. I can't use something that needs to pull in incompatible (with recent ICS) versions of openSSL.

**Reply by viniciussanchez** [reply]
> So to start with RR4D I just use the default and if I find it insufficient I look what the others have to offer?

Yes...

You can implement a driver using ICS and submit your pull request. So I would continue using ICS and RR4D at the same time. Would be great.
